pkg/syncer/storagepool/storagepoolwatch.go

```diff
@@ -0,0 +1,127 @@
+package storagepool
+
+import (
+	"context"
+	"os"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+// SPWatcher watches for events on StoragePool CRD and for each event determines
+// if the events is caused due to external changes to StoragePool or not.
+// If yes then it triggers a reconciliation of StoragePool. It compares generation
+// number of a resource in the event with the generation number stored in
+// SPNameToGenerationNumberMap to determine external change.
+type SPWatcher struct {
+	spWatch                     watch.Interface
+	// A successful update operation to a resource instance increases its generation number by 1.
+	// So all create, delete and update operation to StoragePool should also revise gen num in this map.
+	SPNameToGenerationNumberMap map[string]int64
+}
+
+// InitStoragePoolWatch initialises a SPWatcher instance.
+func InitStoragePoolWatch() *SPWatcher {
```

Comment:
Can we just call this `newStoragePoolWatch()`

Reply:
done :)